test/adapter/json_api/api_objects/relationship_test.rb

```diff
@@ -0,0 +1,168 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        module ApiObjects
+          class RelationshipTest < ActiveSupport::TestCase
+            def setup
+              @blog = Blog.new(id: 1)
+              @author = Author.new(id: 1, name: 'Steve K.', blog: @blog)
+              @serializer = BlogSerializer.new(@blog)
+              ActionController::Base.cache_store.clear
+            end
+
+            def test_relationship_with_data
+              expected = {
+                data: {
+                  id: '1',
+                  type: 'blogs'
+                }
+              }
+              test_relationship(expected, options: { include_data: true })
+            end
+
+            def test_relationship_with_nil_model
+              @serializer = BlogSerializer.new(nil)
+              expected = { data: nil }
+              test_relationship(expected, options: { include_data: true })
+            end
+
+            def test_relationship_with_nil_serializer
+              @serializer = nil
+              expected = { data: nil }
+              test_relationship(expected, options: { include_data: true })
+            end
+
+            def test_relationship_with_data_array
+              posts = [Post.new(id: 1), Post.new(id: 2)]
+              @serializer = ActiveModel::Serializer::ArraySerializer.new(posts)
+              @author.posts = posts
+              @author.blog = nil
+              expected = {
+                data: [
+                  {
+                    id: '1',
+                    type: 'posts'
+                  },
+                  {
+                    id: '2',
+                    type: 'posts'
+                  }
+                ]
+              }
+              test_relationship(expected, options: { include_data: true })
+            end
+
+            def test_relationship_data_not_included
+              test_relationship({}, options: { include_data: false })
```

Comment:
This case should probably raise an error since JSONAPI does not allow the relationship to be empty (http://jsonapi.org/format/#document-resource-object-relationships). I'll let you guys decide how to procede with this one.